code/modules/research/xenobiology/vatgrowing/vatgrower.dm

```diff
@@ -0,0 +1,79 @@
+///Used to make mobs from microbiological samples. Grow grow grow.
+/obj/machinery/plumbing/growing_vat
+	name = "growing vat"
+	desc = "Tastes just like the chef's soup."
+	icon_state = "growing_vat"
+	buffer = 200
+	///List of all microbiological samples in this soup.
+	var/datum/biological_sample/biological_sample
+
+///Add that sexy demnand component
+/obj/machinery/plumbing/growing_vat/Initialize(mapload, bolt)
+	. = ..()
+	AddComponent(/datum/component/plumbing/simple_demand, bolt)
+
+///When we process, we make use of our reagents to try and feed the samples we have.
+/obj/machinery/plumbing/growing_vat/process()
+	if(!is_operational() && !biological_sample)
+		return
```

Comment:
Could you make this start and stop processing depending on if it's operational?

Reply:
While i'm not opposed to this, this sounds like a band-aid that's ultimately better done by a full refactor where we only start processing once it is operational on the base level of machiner, rather than doing this on a child per child basis. Also is_operational is not event based, so there's no reasonable way for me to do this.